core/templates/dev/head/domain/exploration/AnswerGroupObjectFactorySpec.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview unit tests for answer group object factory.
+ */
+
+import { TestBed } from '@angular/core/testing';
+
+import { AnswerGroupObjectFactory } from
+  'domain/exploration/AnswerGroupObjectFactory';
+import { OutcomeObjectFactory } from 'domain/exploration/OutcomeObjectFactory';
+import { RuleObjectFactory } from 'domain/exploration/RuleObjectFactory';
+
+describe('Answer Group Object Factory', () => {
+  let agof = null;
+  let oof = null;
+  let rof = null;
+  let testAnswerGroup = null;
+  let ruleList = null;
+  let backendDict = null;
+  let backendList = null;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [AnswerGroupObjectFactory]
+    });
+    agof = TestBed.get(AnswerGroupObjectFactory);
+    oof = TestBed.get(OutcomeObjectFactory);
+    rof = TestBed.get(RuleObjectFactory);
+    testAnswerGroup = agof.createNew([], oof.createNew('dest_1',
+      'outcome_1', '', []), ['training_data'], 'skill_id-1');
+    ruleList = [
+      rof.createNew('ruleType1', 'input1'),
+      rof.createNew('ruleType2', 'input2')
+    ];
+    backendDict = {
+      rule_specs: [],
```

Comment:
Can you add some rule specs here? This could be why branches are not 100% covered, since toBackendDict() has a branch that works on these.

Reply:
I have added them but it still doesnt seem to change the branch coverage

Reply:
![2](https://user-images.githubusercontent.com/52852877/74937900-06000d80-5413-11ea-96b5-10fe1fd85856.png)